### DISCUSSION TITLE
graph subsumes?

### DISCUSSION THREAD

**Discussion by majidaldo** [pythological/kanren#53] [General]
Q: if i create an ExpressionTuple for say, Or/And, does that subsume lany/lall? In other words, is it possible to just work on expression graphs instead of kanren goals?

related: https://github.com/pythological/kanren/issues/28

**Comment by brandonwillard** [comment]
I'm not sure I understand the Or/And's to which you refer.  Are you talking about using `etuple` evaluation as a means of encapsulating `kanren`'s stream logic?

**Reply by majidaldo** [reply]
Yup

**Reply by brandonwillard** [reply]
It's possible that the generator semantics of `lconj` and `ldisj` could be reproduced using `etuple` evaluation, but I'm not sure.
